keras_nlp/src/tests/test_case.py

```diff
@@ -336,6 +336,32 @@ def run_precision_test(self, cls, init_kwargs, input_data):
                 self.assertEqual(policy.compute_dtype, sublayer.compute_dtype)
                 self.assertEqual(policy.variable_dtype, sublayer.variable_dtype)
 
+    def run_quantization_test(self, cls, init_kwargs, input_data):
+        policy = keras.DTypePolicy("float32")
+        for mode in ["int8", "float8"]:
+            layer = cls(**{**init_kwargs, "dtype": policy})
+            layer.quantize(mode)
+            # Try eager call
+            if isinstance(layer, keras.Model):
+                _ = layer(input_data)
+            elif isinstance(input_data, dict):
+                _ = layer(**input_data)
+            else:
+                _ = layer(input_data)
+            # Verify sublayer's dtype policy
+            for sublayer in layer._flatten_layers():
+                if type(sublayer) is keras.layers.Dense:
+                    self.assertEqual(
+                        f"{mode}_from_float32", sublayer.dtype_policy.name
+                    )
+            # Try saving and reloading the model
+            temp_filepath = os.path.join(self.get_temp_dir(), "layer.keras")
+            layer.save(temp_filepath)
+            reloaded_layer = keras.models.load_model(temp_filepath)
+            self.assertAllClose(
+                layer.predict(input_data), reloaded_layer.predict(input_data)
```

Comment:
Maybe try moving this from `predict()` to a direct call? `predict` will compile, which is slow.

Reply:
Thanks! It's fixed. However, JAX is still slow after the change.

Reply:
Thanks!